src/command_frame.cc

```diff
@@ -1,71 +1,112 @@
 #include <sourcemeta/jsontoolkit/json.h>
 #include <sourcemeta/jsontoolkit/jsonschema.h>
-
 #include <cstdlib>  // EXIT_SUCCESS, EXIT_FAILURE
 #include <iostream> // std::cout
+#include <sstream>  // std::ostringstream
 
 #include "command.h"
 #include "utils.h"
 
-auto intelligence::jsonschema::cli::frame(
-    const std::span<const std::string> &arguments) -> int {
-  const auto options{parse_options(arguments, {})};
-  CLI_ENSURE(!options.at("").empty(), "You must pass a JSON Schema as input")
-  const sourcemeta::jsontoolkit::JSON schema{
-      sourcemeta::jsontoolkit::from_file(options.at("").front())};
-
-  sourcemeta::jsontoolkit::ReferenceFrame frame;
-  sourcemeta::jsontoolkit::ReferenceMap references;
-  sourcemeta::jsontoolkit::frame(schema, frame, references,
-                                 sourcemeta::jsontoolkit::default_schema_walker,
-                                 resolver(options))
-      .wait();
-
-  for (const auto &[key, entry] : frame) {
-    std::cout << "(LOCATION) URI: ";
-    std::cout << key.second << "\n";
-    std::cout << "    Schema           : " << entry.root.value_or("<ANONYMOUS>")
-              << "\n";
-    std::cout << "    Pointer          :";
-    if (!entry.pointer.empty()) {
-      std::cout << " ";
+const char* enumToString(sourcemeta::jsontoolkit::ReferenceEntryType type) {
```

Comment:
You should never return a `const char *`. Please return an `std::string`

Reply:
Also, in C++, we don't use camelCase, but snake_case in variables and function calls. Please do `enum_to_string` or something like that

Reply:
Also, in modern C++, we use trailing return type style. i.e. `auto enum_to_string(...) -> result`

Reply:
Finally, this function doesn't modify its argument, so you want to make the argument `const`

Reply:
Also not strictly necessary in this case, but as a good practice, given this is a private function, please make it `static`

Reply:
Cool! I got it